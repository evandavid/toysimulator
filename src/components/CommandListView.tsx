import React from 'react';
import {View} from 'react-native';
import {ACTIONS} from '../util';
import {CommandText} from './styled';

const CommandListView = ({commands}: {commands: ACTIONS}) => {
  return (
    <View>
      {commands.map((command, index) => (
        <View key={'command' + index}>
          <CommandText>
            {'>>'} {command.type}{' '}
            {command.extraData
              ? `${command.extraData.row}, ${command.extraData.col}, ${command.extraData.f}`
              : ''}
          </CommandText>
        </View>
      ))}
    </View>
  );
};

export default CommandListView;
