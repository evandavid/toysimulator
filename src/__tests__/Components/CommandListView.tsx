import React from 'react';
import renderer from 'react-test-renderer';
import CommandListView from '../../components/CommandListView';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <CommandListView
        commands={[{type: 'PLACE', extraData: {col: 0, row: 0, f: 'EAST'}}]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
