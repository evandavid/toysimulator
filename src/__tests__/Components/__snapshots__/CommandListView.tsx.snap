// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View>
    <Text
      className="sc-hKgJUU gfmGt"
    >
      &gt;&gt;
       
      PLACE
       
      0, 0, EAST
    </Text>
  </View>
</View>
`;
